pkg/controller/hpa/hpa.go

```diff
@@ -73,25 +76,29 @@ func (p *AdvancedHPAController) CreateHPA(ctx context.Context, ahpa *autoscaling
 	return nil
 }
 
-func (p *AdvancedHPAController) NewHPAObject(ahpa *autoscalingapi.AdvancedHorizontalPodAutoscaler) (*autoscalingv2.HorizontalPodAutoscaler, error) {
-	name := fmt.Sprintf("ahpa-%s", ahpa.Name)
+func (p *AdvancedHPAController) NewHPAObject(ctx context.Context, ahpa *autoscalingapi.AdvancedHorizontalPodAutoscaler) (*autoscalingv2.HorizontalPodAutoscaler, error) {
+	metrics, err := p.GetHPAMetrics(ctx, ahpa)
+	if err != nil {
+		return nil, err
+	}
 
+	name := fmt.Sprintf("advancedhpa-%s", ahpa.Name)
 	hpa := &autoscalingv2.HorizontalPodAutoscaler{
 		ObjectMeta: metav1.ObjectMeta{
-			Namespace: ahpa.Namespace, // the same namespace to ahpa
+			Namespace: ahpa.Namespace, // the same namespace to advancedhpa
 			Name:      name,
 			Labels: map[string]string{
 				"app.kubernetes.io/name":                      name,
 				"app.kubernetes.io/part-of":                   ahpa.Name,
-				"app.kubernetes.io/managed-by":                "advanced-hpa-controller",
+				"app.kubernetes.io/managed-by":                known.AdvancedHorizontalPodAutoscalerManagedBy,
```

Comment:
we should use ownerReference to HPA so that we don't need to delete them in controller code.

Reply:
ok, I will remove the deletion code and for the finalizer I want to keep it ,because if we want to do some clean work other than delete hpa and timeseries in future, the finalizer is useful for that.

Reply:
finalizer introduce some difficulties to beginners, as we are open sourcing the project, if someone tear down crane-system namespace first, then the manager will be down, and all apha will be stuck on deletion.